MIGRATION.md

```diff
@@ -0,0 +1,264 @@
+# Migration Guide
+
+Xrpl.js 3.0 reduces bundle size by 60% and completely removes the need for polyfills for browser applications. It does this with minimal breaking changes. It accomplished this by replacing node specific dependencies with ones which would work in the browser.
+
+The primary breaking change is updating all `Buffer` types to `Uint8Array` since `Buffer` is not supported in the browser. `Buffer` objects can usually be directly replaced with `UInt8Array` as `Buffer` extends `Uint8Array`. The primary difference in functionality is that `Buffer` has additional helper functions. We've listed all functions affected by this change below in the `Uint8Array` section.
+
+This migration guide also applies to:
+`ripple-address-codec` 4.3.1 -> 5.0.0
+`ripple-binary-codec` 1.11.0 -> 2.0.0
+`ripple-keypairs` 1.3.1 -> 2.0.0
+
+# Why update to 3.0?
+
+At a high level:
+
+1. 60% filesize reduction
+2. Simplified setup by removing polyfills
+3. Increased reliability
+
+<aside>
+💡 Also, with 3.0, the [reference docs](https://js.xrpl.org/) have received a massive update to include examples, more readable type information, and properly display documentation on functions which are part of the `Client` class.
+</aside>
+
+## 1. 60% size reduction
+
+Through simplifying 3rd party dependencies, xrpl.js is now a much ligher-weight library.
+
+One major source of the project’s large bundle size was polyfills, code that replicated Node specific features in the browser. Instead, we shifted to using 3rd party packages which worked in both Node and the browser by default.
+
+A simple fix was removing `lodash` by using es6 array methods and porting over simple helper utilities.
+
+The last main source of the reduction was simplifying the 5 large number libraries down to one.
+
+Before, we had `decimal.js`, `bit-integer`, `bitnumber.js`, and two versions of `bn.js` due to `elliptic`'s transitive dependency tree.
+
+After, we used `@noble` to replace `elliptic` to have just one version of `bn.js`, and normalized our library to just use `bignumber.js` across the board.
```

Comment:
## 1. 60% Size Reduction

We've significantly reduced the size of xrpl.js by 60%, making it a much lighter-weight library.

A major contributor to the project's large bundle size was the use of polyfills, which replicated Node-specific features in the browser. To address this, we transitioned to utilizing 3rd party packages that inherently supported both Node and browser environments.

One straightforward optimization involved eliminating the use of `lodash` by employing ES6 array methods and migrating simple helper utilities.

Another substantial reduction came from simplifying the five large number libraries down to just one. Previously, we relied on `decimal.js`, `bit-integer`, `bitnumber.js`, and two versions of `bn.js` due to `elliptic`'s transitive dependency tree. However, we streamlined this by adopting `@noble` to replace `elliptic`, resulting in the use of just one version of `bn.js`. This allowed us to standardize our library to use `bignumber.js` consistently across the board.
```suggestion
## 1. 60% size reduction

Through simplifying 3rd party dependencies, xrpl.js is now a much ligher-weight library.

One major source of the project’s large bundle size was polyfills, code that replicated Node specific features in the browser. Instead, we shifted to using 3rd party packages which worked in both Node and the browser by default.

A simple fix was removing `lodash` by using es6 array methods and porting over simple helper utilities.

The last main source of the reduction was simplifying the 5 large number libraries down to one.

Before, we had `decimal.js`, `bit-integer`, `bitnumber.js`, and two versions of `bn.js` due to `elliptic`'s transitive dependency tree.

After, we used `@noble` to replace `elliptic` to have just one version of `bn.js`, and normalized our library to just use `bignumber.js` across the board.
```